### PR TITLE
Add FragmentType enum to help with converting fragment_type field to HDF5 group name

### PR DESCRIPTION
**Pull request by eflumerf**

### CODE REVIEW

**Comment by eflumerf** [comment]
I'll note here that this branch does now create build issues with `dfmodules`

**Review by philiprodrigues — COMMENTED** [review]
I'm wondering a bit about the use of an explicit "invalid" fragment type (rather than, say, throwing an exception when a requested fragment type/string isn't in the map). Is there a particular case that we want to handle with the invalid type?

File: include/dataformats/FragmentHeader.hpp [hunk 1]
```diff
@@ -138,6 +150,57 @@ enum class FragmentErrorBits : size_t
...
 
+/**
+ * @brief This enumeration should list all defined Fragment types
+ */
+enum class FragmentType : fragment_type_t
+{
+  kFakeData = 0,   ///< Data created in dfmodules' FakeDataProducer
+  kReplayData = 1, ///< Data created from frame archive in readout
+  kEmuFEMB = 2,    ///< Data created in WIB's FEMB emulator
+  kUnknown         ///< Used when given a string that does not match any in s_fragment_type_names
```
Comment (philiprodrigues) [inline include/dataformats/FragmentHeader.hpp:161]: Would it make sense to have `kUnknown = max value of fragment_type_t` so its value doesn't change when we add new fragment types?

File: include/dataformats/FragmentHeader.hpp [hunk 2]
```diff
@@ -138,6 +150,57 @@ enum class FragmentErrorBits : size_t
...
+/**
+ * @brief Convert a string to a FragmentType value
+ * @param name Name of the type
+ * @return FragmentType corresponding to given string
+ */
+inline FragmentType
+string_to_fragment_type(std::string name)
+{
+  for (auto& it : s_fragment_type_names) {
+    if (it.second == name)
```
Comment (mroda88) [inline include/dataformats/FragmentHeader.hpp:197]: using find here would be a little more efficient, rather than doing the loop

**Review by philiprodrigues — APPROVED** [review]
LGTM

**Pull request merged**